NSubstitute.DbConnection/NSubstitute.DbConnection.Dapper.Tests/DapperQueryAsyncTests.cs

```diff
@@ -48,4 +49,18 @@ public async Task ShouldMockQueryAsyncUsingRecordType()
         result[0].Key.Should().Be(1);
         result[0].Value.Should().Be("abc");
     }
+
+    [Test]
+    public void ShouldMockQueryAsyncThrow()
+    {
+        var mockConnection = Substitute.For<IDbConnection>().SetupCommands();
+        mockConnection.SetupQuery("delete from table")
+            .Throws(new Exception());
+
+        using var command = mockConnection.CreateCommand();
+        command.CommandText = "delete from table";
+        mockConnection.Open();
+
+        Assert.ThrowsAsync<Exception>(async () => { await mockConnection.QueryAsync("delete from table"); });
```

Comment:
For the sake of consistenly using FluentAssertions, I would use something like this instead here:

```
        var act = async () => await mockConnection.QueryAsync("delete from table");
        await act.Should().ThrowAsync<Exception>();

```

(this allows you to then chain assertions on e.g. the exception message)

```
await act.Should().ThrowAsync<Exception>().Which.Message.Should().Be("foo");
```